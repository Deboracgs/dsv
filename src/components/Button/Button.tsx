import { Button as ButtonMUI } from "@mui/material";

export interface ButtonProps {
  id: string;
  onClick?: () => void;
  label: string;
  variant?: "text" | "outlined" | "contained";
}

const Button = ({ id, onClick, label, variant }: ButtonProps): React.ReactElement => {
  const onClickButton = (e: React.MouseEvent<HTMLElement>) => {
    e.stopPropagation();
    if (typeof onClick === "function") onClick();
  };

  return (
    <ButtonMUI
      id={`button-${id}`}
      data-testid={`button-${id}`}
      variant={variant}
      onClick={(e: React.MouseEvent<HTMLElement>) => onClickButton(e)}
    >
      {label}
    </ButtonMUI>
  );
};

Button.defaultProps = {
  variant: "contained"
};
export default Button;
