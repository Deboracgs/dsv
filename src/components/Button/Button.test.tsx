import { fireEvent, render } from "@testing-library/react";
import WrapperMock from "../../utils/wrapperMock";
import Button, { ButtonProps } from "./Button";

describe("Button Component", () => {
  const defaultProps: ButtonProps = {
    id: "test",
    label: "Label",
    onClick: jest.fn()
  };

  const rendered = (props: ButtonProps) =>
    render(
      <WrapperMock>
        <Button {...props} />
      </WrapperMock>
    );

  it("render", () => {
    const { getByTestId, asFragment } = rendered(defaultProps);

    const element = getByTestId(`button-${defaultProps.id}`);

    expect(element).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it("onClick", () => {
    const onClickSpy = jest.spyOn(defaultProps, "onClick");
    const { getByTestId } = rendered(defaultProps);

    const btnCommon = getByTestId(`button-${defaultProps.id}`);
    fireEvent.click(btnCommon);

    expect(onClickSpy).toHaveBeenCalled();
  });
});
