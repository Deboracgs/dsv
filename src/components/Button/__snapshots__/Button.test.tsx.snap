// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component render 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
    data-testid="button-test"
    id="button-test"
    tabindex="0"
    type="button"
  >
    Label
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
